services/QuillLMS/client/app/bundles/Staff/components/comprehension/rulesAnalysis/ruleAnalysis.tsx

```diff
@@ -90,35 +115,41 @@ const RuleAnalysis = ({ history, match }) => {
   async function massUnmark() { massMark(null)}
 
   async function massMark(rating) {
-    updateResponseStrengthInState(selectedIds, rating)
     setSelectedIds([])
     massCreateOrUpdateFeedbackHistoryRating({ rating, feedback_history_ids: selectedIds}).then((response) => {
-      queryCache.refetchQueries(`rule-feedback-histories-by-rule-${ruleId}`);
+      queryCache.refetchQueries([`rule-feedback-histories-by-rule-${ruleId}-${promptId}`, ruleId, promptId, startDateForQuery, endDateForQuery]);
     });
   }
 
-  function updateResponseStrengthInState(responseIds, rating) {
-    const newResponses = [...responses]
-    responseIds.forEach(id => {
-      const indexOfResponseToChange = responses.findIndex(r => r.response_id === id)
-      const responseToChange = responses[indexOfResponseToChange]
-      responseToChange.strength = rating
-      newResponses[indexOfResponseToChange] = responseToChange
-    })
-    setResponses(newResponses)
-  }
```

Comment:
So the reason we were preserving this in the state wasn't because the refetching didn't work, but because that was too slow - it was specifically requested by the curriculum team that we show the value as soon as it is clicked. I think if we remove this code, it's going to get slow again, but if that doesn't bear out in testing then that's fine with me.

Reply:
so I actually learned recently that for the query refetch to work, you have to pass the full array of args, ie:

```
queryCache.refetchQueries([`rule-feedback-histories-by-rule-${ruleId}-${promptId}`, ruleId, promptId, startDateForQuery, endDateForQuery]);
```

so I don't think this was fetching anything, I think the reason it was slow is because that initial useQuery call checks for new data every like 15 seconds or so. Testing this on staging, the update happens in less than a second